code/modules/mob/living/carbon/carbon.dm

```diff
@@ -98,6 +98,8 @@
 		return 0
 	if(stun)
 		Stun(4)
+	if(ismachineperson(src)) //IPCs do not vomit particulates
```

Comment:
Given they're an IPC it probably shouldn't return the stun in the first place since they can't vomit anyway, the stun itself is there for species that are actually able to vomit. This should probably be at the start of the proc instead to prevent the stun.

I believe IPC booze already has its own stun outside of vomiting(correct me if I'm wrong), but if it doesn't it'd be more appropriate to change something there rather than allow IPC's to suffer any sort of effects from an action they can't perform in the first place.

Reply:
Good point. We talked on discord. Made change and committed it.